### ISSUE TITLE
Not run on newly VS Code 1.53.0

### ISSUE THREAD

**Issue by NormandoHall** [liximomo/vscode-sftp#881]
**Do you read the FAQ?**
- [ x ] Yes.
- [ x ] [I am going to read now.](https://github.com/liximomo/vscode-sftp/blob/master/FAQ.md)

**Describe the bug**
I was update VS Code, and not run anymore this great extension. I need it a lot.

**To Reproduce**
1. Go to sftp configured sites clicking at icon, then sites, and then get the error

**Expected behavior**
Connect

**Screenshots**

![image](https://user-images.githubusercontent.com/2943127/106996961-b53b0f00-6760-11eb-8138-e714985a7dd0.png)

![image](https://user-images.githubusercontent.com/2943127/106996209-3c878300-675f-11eb-8ed2-d986291e5e5f.png)

![image](https://user-images.githubusercontent.com/2943127/106996921-a05e7b80-6760-11eb-86c9-0b506cf2d0fa.png)

**Desktop (please complete the following information):**
Version: 1.53.0 (user setup)
Commit: 8490d3dde47c57ba65ec40dd192d014fd2113496
Date: 2021-02-03T20:36:38.611Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 6.1.7601

**Extension Logs from Startup** - *required*
  1.  Open User Settings.

      * On Windows/Linux - File > Preferences > Settings
      * On macOS - Code > Preferences > Settings
  2. Set `sftp.debug` to `true` and reload vscode.
  3. Reproduce the problem, get the logs from View > Output > sftp.

Nothing from SFTP. Only alert: 

command: "sftp.remoteExplorer.refresh" not found

SFTP version: 1.12.9

**Comment by NormandoHall** [comment]
Seems an issue with latest vsc update:

https://github.com/microsoft/vscode/issues/115818

**Comment by camoralb** [comment]
Same issue on MacOS Big Sur after vscode update. Version Version: 1.53.0

**Comment by wuzhitian** [comment]
Same issue on MacOS 11.2 Big Sur after vscode update. Version: 1.53.0

**Comment by mfunollet** [comment]
Same issue on Linux Mint 20.1 Ulyssa after VScode update to version 1.53.0 :(

**Comment by jasonwilliams** [comment]
So i brought this up with VSCode, they suggested extension bisect. 
It brought me back here:

"Extension Bisect is done and has identified liximomo.sftp as the extension causing the problem."
https://github.com/microsoft/vscode/issues/115859

**Comment by Clinteastman** [comment]
Same issue on Windows 10 20H2 after vscode update. Version: 1.53.0

**Comment by waterdropct** [comment]
Same issue on Windows_NT x64 10.0.18363 after vscode update. Version: 1.53.0

**Comment by apolinux** [comment]
Same issue on Linux Ubuntu 20.04 after vscode update. version 1.53.0

**Comment by minstyle** [comment]
Same issue on Windows 10 Pro 20H2 after vscode update to version 1.53.0.

**Comment by NormandoHall** [comment]
Meanwhile fix this issue, downgrade to 1.52.1 can work again

https://code.visualstudio.com/updates/v1_52

**Comment by drossi70** [comment]
Same issue on Windows 10 Pro 20H2 after vscode update to version 1.53.0.
After install of vascode previous version the extension work.
Disabled the vascode auto-update from setting. Thanks

**Comment by 3b0b** [comment]
I'm asking here because this issue has the most participants, of the seemingly related ones I've found: Has anyone had this issue on a relatively recent target host OS? I had this problem with both this extension and vscode-sshfs on the 1.53 insider build trying to connect to RHEL 6, but not connecting to RHEL 8 (SchoofsKelvin/vscode-sshfs#225). Another user of that extension reported the problem with Debian 4 (SchoofsKelvin/vscode-sshfs#239). Someone reporting what seems like it _might_ be...well, not the root cause, but the underlying commonality (mscdex/ssh2#967) had the issue trying to connect at work, but not connecting to a raspberry (pi, I assume). Not sure there is a significance, but it at least seems _potentially_ relevant.

**Comment by camoralb** [comment]
Yes, it´s happening in my MacOS Big Sur, but only connectiong to my hostgator hosting account. When connecting to some Debian Servers it is working without issues... Although, using cyberduck I can connect to shh without problem to hostgator.

> I'm asking here because this issue has the most participants, of the seemingly related ones I've found: Has anyone had this issue on a relatively recent target host OS? I had this problem with both this extension and vscode-sshfs on the 1.53 insider build trying to connect to RHEL 6, but not connecting to RHEL 8 ([SchoofsKelvin/vscode-sshfs#225](https://github.com/SchoofsKelvin/vscode-sshfs/issues/225)). Another user of that extension reported the problem with Debian 4 ([SchoofsKelvin/vscode-sshfs#239](https://github.com/SchoofsKelvin/vscode-sshfs/issues/239)). Someone reporting what seems like it _might_ be...well, not the root cause, but the underlying commonality ([mscdex/ssh2#967](https://github.com/mscdex/ssh2/issues/967)) had the issue trying to connect at work, but not connecting to a raspberry (pi, I assume). Not sure there is a significance, but it at least seems _potentially_ relevant.

**Comment by borsTiHD** [comment]
> 
> 
> I'm asking here because this issue has the most participants, of the seemingly related ones I've found: Has anyone had this issue on a relatively recent target host OS? I had this problem with both this extension and vscode-sshfs on the 1.53 insider build trying to connect to RHEL 6, but not connecting to RHEL 8 ([SchoofsKelvin/vscode-sshfs#225](https://github.com/SchoofsKelvin/vscode-sshfs/issues/225)). Another user of that extension reported the problem with Debian 4 ([SchoofsKelvin/vscode-sshfs#239](https://github.com/SchoofsKelvin/vscode-sshfs/issues/239)). Someone reporting what seems like it _might_ be...well, not the root cause, but the underlying commonality ([mscdex/ssh2#967](https://github.com/mscdex/ssh2/issues/967)) had the issue trying to connect at work, but not connecting to a raspberry (pi, I assume). Not sure there is a significance, but it at least seems _potentially_ relevant.

Sorry for the poor explanation of my problem. Yes, I can connect successful to a raspberry pi in my private network at home. But in my environment at work, I have the problem that my node process terminates after I tried to build up a connection to a unix server (dont know which host system though... right now...). I hope this information helps. :)

**Comment by tukerty** [comment]
same here!

Extension terminates when connecting to Centos 6 host
Debian 10 host works allright

MacOs - Big Sur
VsCode - 1.53.0
liximomo.sftp - 1.12.9

**Comment by drossi70** [comment]
Confirm error and restart plugin when connection to
Centos 6 and 7
From windows 10

VsCode - 1.53.0
liximomo.sftp - 1.12.9
Thanks

Il dom 7 feb 2021, 08:54 Timur V <notifications@github.com> ha scritto:

**Comment by ruanimal** [comment]
> same here!
> 
> Extension terminates when connecting to Centos 6 host
> Debian 10 host works allright
> 
> MacOs - Big Sur
> VsCode - 1.53.0
> liximomo.sftp - 1.12.9

centos 7 also works allright

**Comment by apolinux** [comment]
> Same issue on Linux Ubuntu 20.04 after vscode update. version 1.53.0

I downgraded to 1.52.1 and worked again, hoping this bug is fixed.

**Comment by chrisfinnigan-S1** [comment]
Also having this issue on 1.53 Ubutu 18.04 (as described above) Reinstalling 1.52 got me working again.

**Comment by arvislacis** [comment]
For other users facing this issue - as the main problem is related to the Electron and createDiffieHellman then you can disable/remove **`"diffie-hellman-group-exchange-sha256"`** algorithm as specified in Wiki - [SFTP only Config](https://github.com/liximomo/vscode-sftp/wiki/SFTP-only-Config#sshcustomparams).

In other words, you need to edit `sftp.json` file and specify following algorithms as supported for each SFTP profile:
```
"algorithms": {
    "kex": [
        "ecdh-sha2-nistp256",
        "ecdh-sha2-nistp384",
        "ecdh-sha2-nistp521",
        "diffie-hellman-group14-sha1"
    ],
    "cipher": [
        "aes128-ctr",
        "aes192-ctr",
        "aes256-ctr",
        "aes128-gcm",
        "aes128-gcm@openssh.com",
        "aes256-gcm",
        "aes256-gcm@openssh.com"
    ],
    "serverHostKey": [
        "ssh-rsa",
        "ecdsa-sha2-nistp256",
        "ecdsa-sha2-nistp384",
        "ecdsa-sha2-nistp521"
    ],
    "hmac": [
        "hmac-sha2-256",
        "hmac-sha2-512",
        "hmac-sha1"
    ]
}
```

^ As you can see **`"diffie-hellman-group-exchange-sha256"`** is not specified. After doing this everything works fine with 1.53.x versions. Not really the best solution but still.

**Comment by NormandoHall** [comment]
@arvislacis thank you so much. Runs like a charm!

**Comment by apolinux** [comment]
> "algorithms": {
>     "kex": [
>         "ecdh-sha2-nistp256",
>         "ecdh-sha2-nistp384",
>         "ecdh-sha2-nistp521",
>         "diffie-hellman-group14-sha1"
>     ],
>     "cipher": [
>         "aes128-ctr",
>         "aes192-ctr",
>         "aes256-ctr",
>         "aes128-gcm",
>         "aes128-gcm@openssh.com",
>         "aes256-gcm",
>         "aes256-gcm@openssh.com"
>     ],
>     "serverHostKey": [
>         "ssh-rsa",
>         "ecdsa-sha2-nistp256",
>         "ecdsa-sha2-nistp384",
>         "ecdsa-sha2-nistp521"
>     ],
>     "hmac": [
>         "hmac-sha2-256",
>         "hmac-sha2-512",
>         "hmac-sha1"
>     ]
> }

question: can I configure this as default for all profiles?

**Comment by arvislacis** [comment]
@apolinux Yes, looks like you can specify "algorithms" array at the beginning of `sftp.json` file and it will be applied for every profile defined in that file. Some progress there :) although looks like you still need to specify this for every project (would be nice if this could be possible in global level too, for all projects).

**Comment by NormandoHall** [comment]
@arvislacis I have an array of objects in `sftp.json` file. Each object is a profile. I added `algorithms` inside each profile and runs ok. When I tried to place `algorithms` outside each profile, as another profile, get error. Also I tried outside main array of `sftp.json` file, and also get errors. Can you give us an example where to place `algorithms` object? Thanks

This is my `sftp.json` file

```
[
  {
    "name": "xxx",
    "etc": "..."
  },
  {
    "name": "yyy",
    "etc": "..."
  }
]
```

**Comment by arvislacis** [comment]
@NormandoHall This works for me:

![2021-02-16_21-18](https://user-images.githubusercontent.com/626982/108110925-a75c8680-709c-11eb-86ed-45dd3db107d5.png)

I use `profiles` object/key to list all available profiles for specific project, as specified at - https://github.com/liximomo/vscode-sftp#profiles.

**Comment by NormandoHall** [comment]
You're right @arvislacis . I was using multiple context. Thank you

**Comment by Bilalh** [comment]
VS Code 1.54.0 (February) fixed the issue for me.

**Comment by minstyle** [comment]
> VS Code 1.54.0 (February) fixed the issue for me. 

After I upgraded to 1.54.1 it fixed the issue for me, too.

**Comment by dziku86** [comment]
Downgrading from node 13.6.0 to 12.21.0 solved the issue for me.
Also implemented @arvislacis https://github.com/liximomo/vscode-sftp/issues/881#issuecomment-778684764 fix.

VSCode 1.55.0-insider
Windows 7 ia32
node.js 12.21.0
